GLAA.Web/Controllers/EligibilityController.cs

```diff
@@ -10,157 +10,225 @@ namespace GLAA.Web.Controllers
+                    ? RedirectToAction("Introduction", "Eligibility")
+                    : ValidateParentAndRedirect(parent, nextPageId);
+            }
+
+            return RedirectToLastAction(FormSection.Eligibility);
+        }
+
+        protected IActionResult CheckParentValidityAndRedirectBack(int submittedPageId)
+        {
+            var email = session.GetString(CurrentPaEmail);
+            var nextPageId = submittedPageId - 1;
+            var parent = accountCreationViewModelBuilder.Build(email);
+
+            return parent == null && nextPageId > 0
+                ? RedirectBackToAction(FormSection.Eligibility, nextPageId)
+                : ValidateParentAndRedirectBack(parent, nextPageId);
+        }
+
+        protected IActionResult ValidateParentAndRedirect(EligibilityViewModel parent, int nextPageId)
+        {
+            parent.Validate();
+            return parent.IsValid
+                ? RedirectToLastAction(FormSection.Eligibility)
+                : RedirectToAction(FormSection.Eligibility, nextPageId);
         }
 
-        protected override string GetViewPath(FormSection section, int id)
+        protected IActionResult ValidateParentAndRedirectBack(EligibilityViewModel parent, int nextPageId)
         {
-            return $"{section.ToString()}.{id}";
+            parent.Validate();
+
+            return nextPageId > 0
+                ? (parent.IsValid
+                    ? RedirectToLastAction(FormSection.Eligibility)
+                    : RedirectBackToAction(FormSection.Eligibility, nextPageId))
+                : RedirectToLastAction(FormSection.Eligibility);
         }
 
         [HttpGet]
         [ImportModelState]
         [Route("Eligibility/Part/{id}")]
-        public ActionResult Eligibility(int id)
+        public ActionResult Eligibility(int id, bool? back = false)
         {
-            var licenceId = session.GetCurrentLicenceId();
+            session.SetLoadedPage(id);
+            var email = session.GetString(CurrentPaEmail);
 
-            var model = licenceApplicationViewModelBuilder.Build<EligibilityViewModel>(licenceId);
+            var model = accountCreationViewModelBuilder.Build(email);
 
-            return GetNextView(id, FormSection.Eligibility, model);            
+            return back.HasValue && back.Value
+                ? GetPreviousView(id, FormSection.Eligibility, model)
+                : GetNextView(id, FormSection.Eligibility, model);
         }
 
-        [HttpGet]
-        public ActionResult WhatDoINeed()
+        public IActionResult Back(int submittedPageId)
         {
-            return View();
+            return CheckParentValidityAndRedirectBack(submittedPageId);
         }
 
         [HttpGet]
-        public ActionResult WhatIsCovered()
+        public ActionResult Introduction()
         {
-            return View();
+            return View("Introduction");
         }
 
         [HttpGet]
-        public ActionResult OverseasBusiness()
```

Comment:
Have these guidance pages been totally culled? I might expect we need to put those docs somewhere still?

Reply:
I've not heard anything about them. Although when they are re-added I suspect they wouldn't be part of the sign up process.